src/charm.py

```diff
@@ -191,9 +187,6 @@ def _on_dbcluster_relation_changed(self, event):
             else:
                 all_bind_addresses = dict()
 
-            if self._stored.all_bind_addresses == all_bind_addresses:
-                return
-
             self._update_config_file(all_bind_addresses)
 
     def _ensure_db_bind_address(self, relation):
```

Comment:
There's one more early return we need to remove in this function for `db_bind_address`

Reply:
I've changed it to always get the db bind address here instead of caching it. Though I'm not clear on what should happen in the case of the multiple addresses error.

Before, if multiple ip addresses were found then we would write the old IP address to the relation config. I have changed it in this PR to *not* write the ip address of the current unit to config (since without saving it in state we no longer have the old IP address). Is this reasonable behavior?

Reply:
We still want to store `last_bind_addresses` as before, because it is used in `_on_collect_status`. The lack of a singular address should make us go into the blocked state, rather than error, because that allows the charm to keep processing other hooks.

It's just the `db_bind_address` check that we remove.

Reply:
Ok, that's what I was doing, for the reason you say. The behavior in the PR is that is goes into blocked state as before, but the behavior I have I changed is this:

When a unit receives multiple bind addresses, it continues to reload config as normal, but it does not write its own IP to config. It logs what happened and will go into blocked because `len(last_bind_addresses)>1` though it can still process other hooks.

Before, it would have have written `_stored.db_bind_address` which would not have been updated (since we early returned on `len(last_bind_addresses)>1`).

Is it alright for the ip not to be present in config when the charm is blocked? Or should I add in some logic to add in the old ip.